lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/Task/TaskRuntime.cs

```diff
@@ -178,8 +186,7 @@ public void Close(byte[] message)
             }
             catch (Exception e)
             {
-                Utilities.Diagnostics.Exceptions.Caught(e, Level.Error, "Error during Close.", Logger);
-                _currentStatus.SetException(e);
+                Utilities.Diagnostics.Exceptions.CaughtAndThrow(e, Level.Error, "Error during Close.", Logger);
```

Comment:
Why? This way the `Exception` propagates upwards naturally, and the Exception in `IFailedEvaluator` will be naturally of the type that was thrown here, instead of having to catch `TaskCloseException` at the outer layer and report `TaskCloseException.InnerException` to `IFailedEvaluator`.